Installing pandas

In [9]:
!pip install pandas

Importing pandas

In [3]:
import pandas as pd

Creating data - There are two core objects in pandas: the DataFrame and the Series.

DataFrame

A DataFrame is a table. It contains an array of individual entries, each of which has a certain value. Each entry corresponds to a row (or record) and a column.

In [6]:
pd.DataFrame({'Yes': [50, 21], 'No': [131, 2]})

Yes   No
0   50  131
1   21    2

In [7]:
pd.DataFrame({'Bob': ['I liked it.', 'It was awful.'], 'Sue': ['Pretty good.', 'Bland.']})

Bob           Sue
0    I liked it.  Pretty good.
1  It was awful.        Bland.

In [8]:
pd.DataFrame({'Men': [35, 50], 'Women':[40, 20]}, index = ['Class A', 'Class B'])

Men  Women
Class A   35     40
Class B   50     20

Series

A Series, by contrast, is a sequence of data values. If a DataFrame is a table, a Series is a list. And in fact you can create one with nothing more than a list. A Series is, in essence, a single column of a DataFrame. So you can assign row labels to the Series the same way as before, using an index parameter. However, a Series does not have a column name, it only has one overall name.

In [11]:
pd.Series([1, 2, 3, 4, 5])

0    1
1    2
2    3
3    4
4    5
dtype: int64

In [13]:
pd.Series([70, 80, 90], index = ['2016', '2017', '2018'])

2016    70
2017    80
2018    90
dtype: int64

In [14]:
pd.Series([70, 80, 90], index = ['2016', '2017', '2018'], name = 'Product A')

2016    70
2017    80
2018    90
Name: Product A, dtype: int64

Reading data files

use the pd.read_csv() function to read the data into a DataFrame

In [17]:
employment_data = pd.read_csv('business-employment-data.csv')

We can use the shape attribute to check how large the resulting DataFrame is.

In [19]:
employment_data.shape

(24124, 14)

So our new DataFrame has 24124 records split across 14 different columns

We can examine the contents of the resultant DataFrame using the head() command, which grabs the first five rows

In [20]:
employment_data.head()

Series_reference   Period  Data_value Suppressed STATUS   UNITS  Magnitude  \
0      BDCQ.SEA1AA  2011.06     80078.0        NaN      F  Number          0   
1      BDCQ.SEA1AA  2011.09     78324.0        NaN      F  Number          0   
2      BDCQ.SEA1AA  2011.12     85850.0        NaN      F  Number          0   
3      BDCQ.SEA1AA  2012.03     90743.0        NaN      F  Number          0   
4      BDCQ.SEA1AA  2012.06     81780.0        NaN      F  Number          0   

                          Subject                            Group  \
0  Business Data Collection - BDC  Industry by employment variable   
1  Business Data Collection - BDC  Industry by employment variable   
2  Business Data Collection - BDC  Industry by employment variable   
3  Business Data Collection - BDC  Industry by employment variable   
4  Business Data Collection - BDC  Industry by employment variable   

  Series_title_1                     Series_title_2 Series_title_3  \
0    Filled jobs  Agriculture, Forestry and Fishing         Actual   
1    Filled jobs  Agriculture, Forestry and Fishing         Actual   
2    Filled jobs  Agriculture, Forestry and Fishing         Actual   
3    Filled jobs  Agriculture, Forestry and Fishing         Actual   
4    Filled jobs  Agriculture, Forestry and Fishing         Actual   

   Series_title_4  Series_title_5  
0             NaN             NaN  
1             NaN             NaN  
2             NaN             NaN  
3             NaN             NaN  
4             NaN             NaN